src/main/scala/mesosphere/marathon/api/v2/json/V2GroupUpdate.scala

```diff
@@ -58,4 +63,28 @@ object V2GroupUpdate {
     V2GroupUpdate(Some(id), if (apps.isEmpty) None else Some(apps), if (groups.isEmpty) None else Some(groups))
   }
   def empty(id: PathId): V2GroupUpdate = V2GroupUpdate(Some(id))
+
+  implicit val v2GroupUpdateValidator: Validator[V2GroupUpdate] = validator[V2GroupUpdate] { group =>
+    group is notNull
+
+    group.version is hasOnlyOneDefinedOption
+    group.scaleBy is hasOnlyOneDefinedOption
+
+    group.id is optional(PathId.validChild(group.id.map(_.canonicalPath(PathId.empty)).getOrElse(PathId.empty)))
+    group.apps is valid
+    group.groups is valid
+  }
+
+  def hasOnlyOneDefinedOption[A <: Product: ClassTag, B]: Validator[A] =
+  new Validator[A] {
+    def apply( product: A ) = {
+      if (product.productIterator.count {
+        case Some(_) => true
+        case _ => false
+      } <= 1)
```

Comment:
It seemed like a valid solution instead of checking `group.version.isDefined` and then calling the `hasOnlyOneDefinedOption`. I did it because I could not simply make an `if(group.version.isDefined) { group.version is hasOnlyOneDefinedOption }`. There is an implicit `and` conjunction between the rules. Is `<=` not a valid work around?